static/app/routes.tsx

```diff
@@ -797,6 +797,16 @@ function buildRoutes() {
               )
           )}
         />
+        <Route
+          path="new-sentry-function/"
```

Comment:
@vpandiarajan20 We should have sentry functions as their own directory like: `https://sentry.io/settings/:orgSlug/developer-settings/sentry-functions/new/`

The reason we should do this is so for the edit Sentry functions view, we can have it at `https://sentry.io/settings/:orgSlug/developer-settings/sentry-functions/:functionSlug/`. If we had it at `https://sentry.io/settings/:orgSlug/developer-settings/functionSlug/`, then we wouldn't know if it was a Sentry function or public/internal integration.